components/dialog/test/dialog-helper.js

```diff
@@ -63,7 +63,7 @@ export async function reset(page, selector) {
 			}
 		});
 	});
-	return page.click('#open');
+	return focusWithKeyboard(page, '#open');
```

Comment:
Part of dialog's "reset" code in between tests involved clicking on this button. But that put the browser back into mouse-mode instead of keyboard-mode, so the first test in each batch was now not getting the focus outline.

Reply:
This is also confusing because the `#open` button actually does nothing, the dialog is opened instead with `dialog.opened = true`. I was running into this when working on dialog and trying to debug some visual-diffs...

Reply:
Haha yeah. The reason this came up now is that dialogs "focus first" logic looks in the content for something to focus on, but then falls back to the focus trap and focuses on it, which would then focus on the "X" button.

Reply:
lol oh no... now there's a bunch of diffs because the open button is in them and it now has focus.

Reply:
I'd be tempted to say this would be a bit cleaner as `page.$eval(selector, (dialog) => dialog.focus());` - that way it doesn't change the state either way. But it should have the same result as long as nothing else is switching to mouse mode.

Reply:
>This is also confusing because the #open button actually does nothing, the dialog is opened instead with dialog.opened = true. 

I think there are at least a couple tests that open and close the dialog, showing that the opener receives the focus upon closing. That might be why some of this is done with an actual opener element rather than just setting the property.

Reply:
I'm. not sure I follow Erik... this code is run in the `beforeEach` of every test to make sure any dialogs opened in the previous test are closed and the page is back and ready to open another dialog.

Reply:
Yep - it's not an issue, I'm just saying that `focus()` would be less opinionated than using `focusWithKeyboard()` which explicitly forces keyboard mode.

That said, the whole keyboard/mouse mode would ideally be reset to the default (keyboard) before every single visual-diff test anyway, to reduce side effects. So ignore me, it's just an observation 😄